infra/build/functions/build_lib.py

```diff
@@ -496,15 +498,31 @@ def get_project_image_steps(  # pylint: disable=too-many-arguments
             'args': ['buildx', 'use', builder_name]
         },
     ])
-    docker_build_arm_step = get_docker_build_step([image],
-                                                  os.path.join(
-                                                      'projects', name),
-                                                  architecture=_ARM64)
+    docker_build_arm_step = get_docker_build_step(
+        [image, _get_unsafe_name(name)],
+        os.path.join('projects', name),
+        architecture=_ARM64)
     steps.append(docker_build_arm_step)
 
+  if (not experiment and not config.testing and
```

Comment:
Do we need to add push as an explicit step? Is it simpler to just add this to https://cloud.google.com/build/docs/build-config-file-schema#images ?

Reply:
I recall max removing this step because it failed. I think this way is more performant (it runs in the background) and error-tolerant, but I'd understand if you want the simplicity instead.
I do think all of this builder code may be getting too complicated, too many special cases.

Reply:
I recall max removing this step because it failed. I think this way is more performant (it runs in the background) and error-tolerant, but I'd understand if you want the simplicity instead.
I do think all of this builder code may be getting too complicated, too many special cases.

Reply:
Yeah, simplifcity is my main motivation -- this is getting way too complicated.

Reply:
Got it. I think the failure is problematic though because it can lead to issues filed against users.

Reply:
It's been a while so hopefully this is handled better by Cloud Build now. But yeah, good point. If we run into this again your solution is probably the more practical one.

Reply:
I'm low on cycles so let's land this for now and fix it later. I'll file an issue so I don't forget to fix this (though i have a feeling batch isn't dealing with this better).

Reply:
I think this may have broken trial builds?

From https://github.com/google/oss-fuzz/pull/12869 

```sh
Step #0: Step 6/6 : ENTRYPOINT ["python3", "/opt/oss-fuzz/infra/build/functions/ci_trial_build.py"]
Step #0:  ---> Running in 41f524716b04
Step #0: Removing intermediate container 41f524716b04
Step #0:  ---> 7f2a352668c0
Step #0: Successfully built 7f2a352668c0
Step #0: Successfully tagged gcr.io/oss-fuzz-base/trial-build:latest
Finished Step #0
Starting Step #1
Step #1: Already have image: gcr.io/oss-fuzz-base/trial-build
Step #1: INFO:root:Command: ['jvm', '--fuzzing-engines', 'libfuzzer', '--repo', 'https://github.com/google/oss-fuzz', '--branch', 'jazzer-upgrade'].
Step #1: INFO:root:Timeout: 25200, trial build end time: 2024-12-19 17:12:15.050736
Step #1: Creating temporary archive of 5983 file(s) totalling 10.5 MiB before compression.
Step #1: Some files were not included in the source upload.
Step #1: 
Step #1: Check the gcloud log [/builder/home/.config/gcloud/logs/2024.12.19/10.12.16.427915.log] to see which files and the contents of the
Step #1: default gcloudignore file used (see `$ gcloud topic gcloudignore` to learn
Step #1: more).
Step #1: 
Step #1: Uploading tarball of [.] to [gs://oss-fuzz-base_cloudbuild/source/1734603136.91608-bbf8888423974f0bb0f1795ecbf50a72.tgz]
Step #1: ERROR: (gcloud.builds.submit) INVALID_ARGUMENT: invalid build: invalid .steps field: build step #1 - "build-x86_64": the ID is not unique
Step #1: Traceback (most recent call last):
Step #1:   File "/opt/oss-fuzz/infra/build/functions/ci_trial_build.py", line 94, in <module>
Step #1:     sys.exit(main())
Step #1:   File "/opt/oss-fuzz/infra/build/functions/ci_trial_build.py", line 87, in main
Step #1:     result = exec_command_from_github(pull_request_number, repo, branch)
Step #1:   File "/opt/oss-fuzz/infra/build/functions/ci_trial_build.py", line 78, in exec_command_from_github
Step #1:     return trial_build.trial_build_main(command, local_base_build=False)
Step #1:   File "/opt/oss-fuzz/infra/build/functions/trial_build.py", line 379, in trial_build_main
Step #1:     build_and_push_test_images.gcb_build_and_push_images(test_image_suffix)
Step #1:   File "/opt/oss-fuzz/infra/build/functions/build_and_push_test_images.py", line 109, in gcb_build_and_push_images
Step #1:     _run_cloudbuild(build_body)
Step #1:   File "/opt/oss-fuzz/infra/build/functions/build_and_push_test_images.py", line 73, in _run_cloudbuild
Step #1:     subprocess.run([
Step #1:   File "/usr/lib/python3.10/subprocess.py", line 526, in run
Step #1:     raise CalledProcessError(retcode, process.args,
Step #1: subprocess.CalledProcessError: Command '['gcloud', 'builds', 'submit', '--project=oss-fuzz-base', '--config=/opt/oss-fuzz/cloudbuild.yaml']' returned non-zero exit status 1.
Finished Step #1
ERROR
```

Reply:
I'm not 100% where the error comes from, but commented here since the issue seems to come from `get_docker_build_step`

Reply:
Hmm...`Step #1: ERROR: (gcloud.builds.submit) INVALID_ARGUMENT: invalid build: invalid .steps field: build step #1 - "build-x86_64": the ID is not unique`
this doesn't look related but I suppose not much else changed

Reply:
Oh actually, maybe I added the id